File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var numberOfSubarrays = function (nums, k) {
    let cnt = new Array(nums.length + 1).fill(0);
    let t = 0, ans = 0;
    cnt[0] = 1;
    for (let v of nums) {
        t += v & 1;
        ans += t - k >= 0 ? cnt[t - k] : 0;
        cnt[t]++;
    }
    return ans;
};

File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        count = [0] * (len(nums) + 1)
        count[0] = 1
        total, res = 0, 0

        for v in nums:
            total += v & 1
            if total - k >= 0:
                res += count[total - k]
            count[total] += 1

        return res
